### ISSUE TITLE
Multi-line comment detected within single line comment (c++).

### ISSUE THREAD

**Issue by marcel303** [BalazsJako/ImGuiColorTextEdit#52]
Hi,

The following code has incorrect syntax highlighting. One can verify this by copy-pasting it into the editor.

```
	if (hasValue == false && nodeId != kGraphNodeIdInvalid)
	{
		//logDebug("reset realTimeSocketCapture");
		//*this = GraphEdit_Visualizer();
	}
	
	//
	
	const bool hasChannels = channelData.hasChannels();
	
	if (hasChannels)
	{
```

I investigated why, and it's caused by the multi-line comment highlighter running before single line comments are detected. It causes the '/*' in '//*this' above to be interpreted as the start of a multi-line comment. While it should be ignored as it's part of a single line comment. A possible solution would be to detect single line comments together with multi-line ones.

Perhaps eventually to make (multi-line) comments work well across different languages we'll need a separate comment highlighter for each language.

**Comment by BalazsJako** [comment]
Fixed in 8fbb932.

**Comment by marcel303** [comment]
Thanks, seems to work well!
